### ISSUE TITLE
PlotSquared is not saving all flags

### ISSUE THREAD

**Issue by yannicklamprecht** [IntellectualSites/PlotSquared#2686]
# Bug Report Template:

## Required Information section:
> ALL FIELDS  IN THIS SECTION ARE REQUIRED, and must contain appropriate information
### Server type:
**Select one**

- [x] Spigot / Paper *(CraftBukkit should not be used, re-test with Spigot first!)*
- [] Sponge *- NOTE: NOT ACTIVELY MAINTAINED*
- [] NukkitX *- NOTE: NOT ACTIVELY MAINTAINED*

### Minecraft Version:
**Select one**

- [x] Minecraft 1.15
- [] Minecraft 1.14.4
- [] Minecraft 1.13.2
- [] Minecraft Java Edition *other versions, please specify*:
- [] Minecraft Bedrock Edition *specify version*:

### Server build info: 

```
[13:44:41 INFO]: This server is running Paper version git-Paper-66 (MC: 1.15.2) (Implementing API version 1.15.2-R0.1-SNAPSHOT)
```
Version doesn't matter since it's a database issue.

### WorldEdit/FAWE versions:

- [] FAWE *version*:
- [] WorldEdit *version*: 

### Description of the problem:

Each Plot does have Plot_Settings these includes a column flags which is limited to varchar(512).
During addition of many flags and many use_flag values the limit is exceeded.

The dirty fix would be to increase the limitation to 1k or more. The more future proof fix would be to migrate to a better database structure. See Database 1./2./3. Normalform.

| id | plot_forreign_key | flag_name | flag_value |
| --- | ------------- | ----- | ---- |
| 1 | 1 | use | stone_button |
| 2 | 1 | use | oak_door | 
| 3 | 1 | game_mode | creative |

flag_name, flag_value should be considered unique as a whole.
### How to replicate:

E.g. add use flags as NamespaceKeys, like that`minecraft:stone`. Add multiple of them until you exceed the 512 chars. So around 34 Materials or when not using `minecraft:` add an average of 64 Materials.

## Additional Information:
> The information here is optional for you to provide, however it may help us to more readily diagnose any compatibility and bug issues.

### Relevant console output, log lines, and/or screenshots:

```
[13:27:09 INFO]: ========================================                
[13:27:11 INFO]: ============ DATABASE ERROR ============                                                               
[13:27:11 INFO]: There was an error updating the database.               
[13:27:11 INFO]:  - It will be corrected on shutdown                                                                    
[13:27:11 INFO]: ========================================                
[13:27:11 WARN]: java.sql.BatchUpdateException: Data truncation: Data too long for column 'flags' at row 1              
[13:27:11 WARN]:        at sun.reflect.GeneratedConstructorAccessor109.newInstance(Unknown Source)
[13:27:11 WARN]:        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[13:27:11 WARN]:        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
[13:27:11 WARN]:        at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)                                         
[13:27:11 WARN]:        at com.mysql.jdbc.Util.getInstance(Util.java:408)
[13:27:11 WARN]:        at com.mysql.jdbc.SQLError.createBatchUpdateException(SQLError.java:1154)                       
[13:27:11 WARN]:        at com.mysql.jdbc.PreparedStatement.executeBatchSerially(PreparedStatement.java:1832)
[13:27:11 WARN]:        at com.mysql.jdbc.PreparedStatement.executeBatchInternal(PreparedStatement.java:1316)           
[13:27:11 WARN]:        at com.mysql.jdbc.StatementImpl.executeBatch(StatementImpl.java:954)
[13:27:11 WARN]:        at com.github.intellectualsites.plotsquared.plot.database.SQLManager$UniqueStatement.execute(SQLManager.java:3099)
[13:27:11 WARN]:        at com.github.intellectualsites.plotsquared.plot.database.SQLManager.sendBatch(SQLManager.java:392)
[13:27:11 WARN]:        at com.github.intellectualsites.plotsquared.plot.database.SQLManager.lambda$new$0(SQLManager.java:163)
[13:27:11 WARN]:        at org.bukkit.craftbukkit.v1_15_R1.scheduler.CraftTask.run(CraftTask.java:84)
[13:27:11 WARN]:        at org.bukkit.craftbukkit.v1_15_R1.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:54)         
[13:27:11 WARN]:        at com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22)
[13:27:11 WARN]:        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)              
[13:27:11 WARN]:        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[13:27:11 WARN]:        at java.lang.Thread.run(Thread.java:748)                                                        
[13:27:11 WARN]: Caused by: com.mysql.jdbc.MysqlDataTruncation: Data truncation: Data too long for column 'flags' at row 1
[13:27:11 WARN]:        at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3931)                                   
[13:27:11 WARN]:        at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3869)
[13:27:11 WARN]:        at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2524)                                        
[13:27:11 WARN]:        at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2675)
[13:27:11 WARN]:        at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2465)                              
[13:27:11 WARN]:        at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:1912)
[13:27:11 WARN]:        at com.mysql.jdbc.PreparedStatement.executeUpdateInternal(PreparedStatement.java:2133)          
[13:27:11 WARN]:        at com.mysql.jdbc.PreparedStatement.executeBatchSerially(PreparedStatement.java:1810)
[13:27:11 WARN]:        ... 11 more      
```

### Additional relevant comments/remarks:

# AFFIRMATION OF COMPLETION:

- [x] I included all information required in the sections above
- [x] I made sure there are no duplicates of this report [(Use Search)](https://github.com/IntellectualSites/PlotSquared/issues?utf8=%E2%9C%93&q=is%3Aissue)
- [x] I made sure I am using an up-to-date version of PlotSquared
- [x] I made sure the bug/error is not caused by any other plugin
- [] I didn't read but checked everything above.

**Comment by yannicklamprecht** [comment]
TLDR; Database structure needs reconsideration. Plot debug paste is not necessary since it can be traced back to database only issue.

**Comment by Sauilitired** [comment]
Plot flags are being completely rewritten in v5, along with the storage.

**Comment by N0tMyFaultOG** [comment]
The issue has been fixed in PlotSquared v5.
